### ISSUE TITLE
[SelectDialog]: Not able to change the search icon to decorative.

### ISSUE THREAD

**Issue by raoakash01** [SAP/ui5-webcomponents#12009]
### Bug Description

In SelectDialog, we have a search input and search icon with it.
I have received an accessibility issue that the icon should be decorative and I am not able to find any property exposed to do so.

<img width="354" height="356" alt="Image" src="https://github.com/user-attachments/assets/bb1dfba8-1815-4168-8f9d-35a32d33d2fe" />

### Expected Behaviour

the search icon should be marked as a decorative image and should not be focusable or function as a button.

### Isolated Example

https://stackblitz.com/edit/github-ybxm6uqu?file=package.json,src%2FApp.tsx

### Steps to Reproduce

1.
2.
3.
...

### Priority

High

### UI5 Web Components Version

2.7.0

### Browser

Chrome

### Declaration

- [x] I’m not disclosing any internal or sensitive information.
